### ISSUE TITLE
`unused_async` lint does not work for methods

### ISSUE THREAD

**Issue by david-perez** [rust-lang/rust-clippy#9024]
### Summary

This program:

```rust
use async_trait::async_trait;

#[async_trait]
trait AsyncTrait {
    async fn async_baz(&self) -> &'static str;
}

struct App;

impl App {
    pub async fn method(&self) -> &'static str {
        "incorrectly not caught by clippy"
    }
}

#[async_trait]
impl AsyncTrait for App {
    async fn async_baz(&self) -> &'static str {
        "correctly not caught by clippy"
    }
}

pub async fn freestanding() -> &'static str {
    "correctly caught by clippy"
}

fn main() {
    let app = App;
    let _ = app.method();
}
```

Makes `cargo clippy -- --deny clippy::unused_async` yield:

```
error: unused `async` for function with no await statements
  --> src/main.rs:23:1
   |
23 | / pub async fn freestanding() -> &'static str {
24 | |     "correctly caught by clippy"
25 | | }
   | |_^
   |
   = note: requested on the command line with `-D clippy::unused-async`
   = help: consider removing the `async` from this function
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unused_async

error: could not compile `clippy-unused-async` due to previous error
```

However, I see no reason why it should not lint against `method`.

Lint introduced in https://github.com/rust-lang/rust-clippy/pull/7225.

### Version

```text
rustc 1.61.0 (fe5b13d68 2022-05-18)
binary: rustc
commit-hash: fe5b13d681f25ee6474be29d748c65adcd91f69e
commit-date: 2022-05-18
host: x86_64-unknown-linux-gnu
release: 1.61.0
LLVM version: 14.0.0
```
